File: gcc/testsuite/gcc.dg/tree-ssa/inline-1.c
/* { dg-do compile } */ 
/* { dg-options "-O2 -fdump-tree-optimized" } */


typedef struct {
  double min;
  double max;
} interval;
inline interval add(interval x, interval y)  __attribute__((always_inline));
inline interval add(interval x, interval y)  
{
  interval r;
  r.min = x.min + y.min;
  r.max = x.max + y.max;
  return r;
}
interval foo (interval a, interval b, interval c)
{
  return add (a, add (b, c));
}


/* { dg-final { scan-tree-dump-times "\\(struct interval\\)" 0 "optimized"} } */
/* { dg-final { cleanup-tree-dump "optimized" } } */

File: gcc/testsuite/rust/compile/inline_1.rs
// { dg-additional-options "-fdump-tree-gimple" }
#[inline]
fn test_a() {}

// { dg-final { scan-tree-dump-times {always_inline} 1 gimple } }
#[inline(always)]
fn test_b() {}

#[inline(never)]
fn test_c() {}

fn main() {
    test_a();
    test_b();
    test_c();
}
